components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -560,6 +566,64 @@ public static boolean authenticateClient(String clientId, String clientSecretPro
         return true;
     }
 
+    /**
+     * Authenticate the OAuth Consumer.
+     *
+     * @param clientId             Consumer Key/ Id.
+     * @param clientSecretProvided Consumer Secret issued during the time of registration.
+     * @param appTenant            Tenant domain of the application.
+     * @return true, if the authentication is successful, false otherwise.
+     * @throws IdentityOAuthAdminException Error when looking up the credentials from the database
+     */
+    public static boolean authenticateClient(String clientId, String clientSecretProvided, String appTenant)
```

Comment:
- Deprecate the existing method and inside that method call the new method.
- Don't duplicate the method body.